homeassistant/components/notify/homematic.py

```diff
@@ -52,23 +53,11 @@ def __init__(self, hass, data):
 
     def send_message(self, message="", **kwargs):
         """Send a notification to the device."""
-        attr_data = kwargs.get(ATTR_DATA)
-        if attr_data is not None:
-            if 'address' in attr_data:
-                self.data[ATTR_ADDRESS] = attr_data['address']
-            if 'channel' in attr_data:
-                self.data[ATTR_CHANNEL] = attr_data['channel']
-            if 'param' in attr_data:
-                self.data[ATTR_PARAM] = attr_data['param']
-            if 'value' in attr_data:
-                self.data[ATTR_VALUE] = attr_data['value']
-            if 'interface' in attr_data:
-                self.data[ATTR_INTERFACE] = attr_data['interface']
+        data = copy.deepcopy(self.data)
```

Comment:
A deepcopy is not needed as `update` is not recursive. This should do the trick:

```python
data = {**self.data, **kwargs.get(ATTR_DATA, {})}
```